jobbergate-api/jobbergate_api/apps/job_script_templates/routers.py

```diff
@@ -93,99 +87,81 @@ async def job_script_template_get(
-    sort_field: Optional[str] = Query(None),
+    search: str | None = Query(None),
+    sort_field: str | None = Query(None),
     sort_ascending: bool = Query(True),
-    token_payload: TokenPayload = Depends(guard.lockdown(Permissions.JOB_TEMPLATES_VIEW)),
-    service: TableService = Depends(template_service),
+    secure_session: SecureSession = Depends(
+        secure_services(Permissions.JOB_TEMPLATES_VIEW, services=[crud_service])
+    ),
 ):
     """Get a list of job script templates."""
     logger.debug("Preparing to list job script templates")
 
-    identity_claims = IdentityClaims.from_token_payload(token_payload)
-
-    list_kwargs: dict[str, Any] = dict(
+    return await crud_service.paginated_list(
+        include_null_identifier=include_null_identifier,
+        user_email=secure_session.identity_payload.email if user_only else None,
         search=search,
         sort_field=sort_field,
         sort_ascending=sort_ascending,
     )
 
-    if user_only:
-        list_kwargs["owner_email"] = identity_claims.email
-    if not include_null_identifier:
-
-        def custom_filter(table, query):
-            return query.filter(not_(table.identifier.is_(None)))
-
-        list_kwargs["custom_filter"] = custom_filter
-
-    return await service.list(**list_kwargs)
-
 
 @router.put(
     "/{id_or_identifier}",
     status_code=status.HTTP_200_OK,
     description="Endpoint to update a job script template by id or identifier",
     response_model=JobTemplateResponse,
-    dependencies=[Depends(guard.lockdown(Permissions.JOB_TEMPLATES_EDIT))],
+    dependencies=[Depends(secure_services(Permissions.JOB_TEMPLATES_EDIT, services=[crud_service]))],
 )
 async def job_script_template_update(
     update_request: JobTemplateUpdateRequest,
     id_or_identifier: int | str = Path(),
-    service: TableService = Depends(template_service),
 ):
     """Update a job script template by id or identifier."""
     logger.info(f"Updating job script template {id_or_identifier=} with {update_request=}")
-    await service.update(id_or_identifier, **update_request.dict(exclude_unset=True))
-    return await service.get(id_or_identifier)
+    return await crud_service.update(id_or_identifier, **update_request.dict(exclude_unset=True))
 
 
 @router.delete(
     "/{id_or_identifier}",
     status_code=status.HTTP_204_NO_CONTENT,
     description="Endpoint to delete a job script template by id or identifier",
-    dependencies=[Depends(guard.lockdown(Permissions.JOB_TEMPLATES_EDIT))],
+    dependencies=[Depends(secure_services(Permissions.JOB_TEMPLATES_EDIT, services=[crud_service]))],
 )
 async def job_script_template_delete(
     id_or_identifier: int | str = Path(),
-    service: TableService = Depends(template_service),
 ):
     """Delete a job script template by id or identifier."""
     logger.info(f"Deleting job script template with {id_or_identifier=}")
-    await service.delete(id_or_identifier)
+    await crud_service.delete(id_or_identifier)
```

Comment:
When a template or job script is deleted, it cascades to their files on the database, but the objects are still on s3. They will be garbage collected, but we should consider creating a `background_tasks` to handle it.

Placeholder: https://app.clickup.com/t/18022949/PENG-1544

Reply:
Thank you. That's a good tech debt item.

Reply:
Addressed in commit a59185b